### PR TITLE
Table filters - Fix date filter issue

### PR DESCRIPTION
**Pull request by Ricokola**
Related to: WET-214

Fixing the issue when filtering using a date from the date-picker for which there is no data available and still the search results is providing data that are dated between the 10 and 19 of every months.

### CODE REVIEW

**Review by duboisp — CHANGES_REQUESTED** [review]
as discussed, please apply the change for when the field value is retrieved in the other conditional, like number filter. And, as you mentioned, update the other default that seems to present the same kind of issue.

**Comment by GormFrank** [comment]
To be tested locally and insure French working example also benefits from the extra entries in the table.

File: src/plugins/tables/tables.js [hunk 1]
```diff
@@ -253,17 +253,17 @@ $document.on( "submit", ".wb-tables-filter", function( event ) {
 		if ( $elm.is( "select" ) ) {
 			$value = $elm.find( "option:selected" ).val();
 		} else if ( $elm.is( "input[type='number']" ) ) {
-			var $val = $elm.val(), $minNum, $maxNum;
```
Comment (duboisp) [inline src/plugins/tables/tables.js:256]: Please keep the variable caching.

**Review by duboisp — CHANGES_REQUESTED** [review]

**Review by duboisp — CHANGES_REQUESTED** [review]
Change line 239, for number, similarly as you did for date.

File: src/plugins/tables/tables.js [hunk 2]
```diff
@@ -253,12 +254,12 @@ $document.on( "submit", ".wb-tables-filter", function( event ) {
...
+			var $minNum, $maxNum;
 
 			// Retain minimum number (always the first number input)
 			if ( $cachedVal === "" ) {
 				$cachedVal = parseFloat( $val );
-				$cachedVal = ( $cachedVal ) ? $cachedVal : "-0";
+				$cachedVal = ( $cachedVal ? $cachedVal : $val );
```
Comment (duboisp) [inline src/plugins/tables/tables.js:262]: Humm, we will need to better understand the use of "-0" here and how it used later. May be we can look at the history to find out more information about the use case of using the value "-0"
 
If the value of in the input don't parse, then we might just fallback on a simple text search matching the input value.

Comment (thekodester) [reply src/plugins/tables/tables.js:262]: I originally coded a lot of these plugins. Quite a few of these changes are completely unrelated to the date filter and are related the the number filter.

Does the original poster have example data that can be used to show when the date filter breaks. If so I can see if I’m able to run some test.

Comment (thekodester) [reply src/plugins/tables/tables.js:262]: Yes for the date filter function yes I can see that issue. But the code you are changing on this line is unrelated to the date filter.

**Review by duboisp — APPROVED** [review]

**Review by GormFrank — APPROVED** [review]

**Pull request merged**